Sources/Commands/SwiftPackageCollectionsTool.swift

```diff
@@ -0,0 +1,358 @@
+
+        @Argument(help: "URL of the collection to describe")
+        var collectionUrl: String?
+
+        @OptionGroup
+        var profileOptions: ProfileOptions
+
+        mutating func run() throws {
+            guard let collectionUrlString = collectionUrl, let collectionUrl = URL(string: collectionUrlString) else {
+                throw CollectionsError.missingArgument("collectionUrl")
+            }
+
+            let collections = try await { self.collections.listCollections(identifiers: nil, in: profileOptions.usedProfile, callback: $0) }
+            let source = PackageCollectionsModel.CollectionSource(url: collectionUrl)
+
+            guard let collection = collections.first(where: { $0.source == source }) else {
+                throw CollectionsError.noCollectionMatchingURL(collectionUrlString)
+            }
+
+            let description = optionalRow("Description", collection.description)
+            let keywords = optionalRow("Keywords", collection.keywords?.joined(separator: ", "))
+            let createdAt = DateFormatter().string(from: collection.createdAt)
+            let packages = collection.packages.map { "\($0.repository.url)" }.joined(separator: "\n")
+
+            print("""
+                Name: \(collection.name)
+                Source: \(collection.source.url)\(description)\(keywords)
+                Created At: \(createdAt)
+                Packages:
+                \(packages)
+            """)
+        }
+    }
+
+    // MARK: Search
+
+    enum SearchMethod: String, EnumerableFlag {
+        case keywords
+        case module
+    }
+
+    struct Search: ParsableCommand {
+        static var configuration = CommandConfiguration(abstract: "Search for packages by keywords or module names")
+
+        @OptionGroup
+        var jsonOptions: JSONOptions
+
+        @OptionGroup
+        var profileOptions: ProfileOptions
+
+        @Flag(help: "Pick the method for searching")
+        var searchMethod: SearchMethod
+
+        @Argument(help: "Search query")
+        var searchQuery: String?
+
+        mutating func run() throws {
+            guard let searchQuery = searchQuery else {
+                throw CollectionsError.missingArgument("searchQuery")
+            }
+
+            switch searchMethod {
+            case .keywords:
+                let results = try await { collections.findPackages(searchQuery, collections: nil, profile: profileOptions.usedProfile, callback: $0) }
+
+                results.items.forEach {
+                    print("\($0.package.repository.url): \($0.package.description ?? "")")
+                }
+
+            case .module:
+                let results = try await { collections.findTargets(searchQuery, searchType: .exactMatch, collections: nil, profile: profileOptions.usedProfile, callback: $0) }
+
+                results.items.forEach {
+                    $0.packages.forEach {
```

Comment:
this may show duplicates since we are not grouping by collection so maybe flatten into a set?

Reply:
also today this will search across *all* targets not just modules. should we change the API so it allows searching only on modules?

Reply:
That seems very reasonable.

Reply:
I'm not sure I follow, what would be a target that isn't a module?

Reply:
A test target is generally not an importable module.

Reply:
It might be technically importable but doesn't seem like the kind of thing that should be offered in a `import` completion.

Reply:
if we wanted to avoid the confusion from the term "module" vs " target" we can have the API take a targetType filter?

Reply:
We could also rephrase in terms of "library target"

Reply:
Are binary targets considered to be libraries?

Reply:
I would consider any target that's part of a library product a library target, so binaries as well.

Reply:
Sounds good.  Rule of thumb should be "anything that can be `import`ed", I think.